types/conv.go

```diff
@@ -96,36 +92,20 @@ func ToABCIBlockMeta(block *Block) (*tmtypes.BlockMeta, error) {
 // ToABCICommit converts Dymint commit into commit format defined by ABCI.
 // This function only converts fields that are available in Dymint commit.
 // Other fields (especially ValidatorAddress and Timestamp of Signature) has to be filled by caller.
-func ToABCICommit(commit *Commit, header *Header) *tmtypes.Commit {
-	headerHash := header.Hash()
+func ToABCICommit(commit *Commit) *tmtypes.Commit {
 	tmCommit := tmtypes.Commit{
 		Height: int64(commit.Height), //nolint:gosec // height is non-negative and falls in int64
 		Round:  0,
 		BlockID: tmtypes.BlockID{
-			Hash: headerHash[:],
+			Hash: commit.HeaderHash[:],
 			PartSetHeader: tmtypes.PartSetHeader{
 				Total: 1,
-				Hash:  headerHash[:],
+				Hash:  commit.HeaderHash[:],
 			},
 		},
 	}
-	// Check if TMSignature exists. if not use the previous dymint signature for backwards compatibility.
-	if len(commit.TMSignature.Signature) == 0 {
-		for _, sig := range commit.Signatures {
-			commitSig := tmtypes.CommitSig{
-				BlockIDFlag: tmtypes.BlockIDFlagCommit,
-				Signature:   sig,
-			}
-			tmCommit.Signatures = append(tmCommit.Signatures, commitSig)
-		}
-		// This assumes that we have only one signature
-		if len(commit.Signatures) == 1 {
-			tmCommit.Signatures[0].ValidatorAddress = header.ProposerAddress
-			tmCommit.Signatures[0].Timestamp = header.GetTimestamp()
-		}
-	} else {
-		tmCommit.Signatures = append(tmCommit.Signatures, commit.TMSignature)
-	}
-
+	// Very old commits may not have a TM signature already. Should run a prior version of
+	// dymint to handle them.
+	tmCommit.Signatures = append(tmCommit.Signatures, commit.TMSignature)
 	return &tmCommit
```

Comment:
I think it makes sense to remove this, since
- I guess the only blocks without the signatures are very old? Are there even any on mainnet?
- Nowadays you need to be running the right dymint to process old blocks anyway